llvmlite/ir/values.py

```diff
@@ -14,13 +15,15 @@
 
 _VALID_CHARS = (frozenset(map(ord, string.ascii_letters)) |
                 frozenset(map(ord, string.digits)) |
-                frozenset('._-$'))
+                frozenset(map(ord, ' !#$%&\'()*+,-./:;<=>?@[]^_`{|}~')))
 
 
 def _escape_string(text, _map={}):
     """
     Escape the given bytestring for safe use as a LLVM array constant.
     """
+    if isinstance(text, six.text_type):
```

Comment:
I don't see why. This code is common for all double-quoted strings in the IR, whether originally strings (and thus user-facing UTF-8 identifiers) or byte arrays (probably emitted by the compiler).